data/spells/scripts/monster/wyrm_wave.lua

```diff
@@ -0,0 +1,17 @@
+area = {
+	{0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 0},
+	{0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0},
+	{0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0},
+	{0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 3, 0, 0, 0, 0, 0}
+}
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_ENERGYDAMAGE)
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_PURPLEENERGY)
+combat:setArea(createCombatArea(area))
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
+end
```

Comment:
this spell not needed in lua, just use the one that is already in wyrm file and adjust the damage:

```
		<attack name="energy" interval="2000" chance="10" length="5" spread="2" min="-130" max="-200">
			<attribute key="areaEffect" value="purpleenergy" />
		</attack>
```